packages/framework/src/validators/class-validator.validator.ts

```diff
@@ -0,0 +1,133 @@
+function addAdditionalPropertiesFalse(schema: any): JsonSchema {
+  if (schema && typeof schema === 'object' && schema?.type === 'object') {
+    // eslint-disable-next-line no-param-reassign
+    schema.additionalProperties = false;
+  }
+
+  if (schema.properties) {
+    for (const key in schema.properties) {
+      if (schema.properties.hasOwnProperty(key)) {
+        addAdditionalPropertiesFalse(schema.properties[key]);
+      }
+    }
+  }
+
+  return schema;
+}
+function formatErrors(errors: ValidationError[], parentPath = ''): { path: string; message: string }[] {
+  return errors.flatMap((err) => {
+    const currentPath = `${parentPath}/${err.property}`.replace(/\/+/g, '/');
+
+    if (err.children && err.children.length > 0) {
+      // Recursively format the children
+      return formatErrors(err.children, currentPath);
+    } else {
+      // Base case: no children, return the formatted error
+      return {
+        path: currentPath,
+        message: Object.values(err.constraints || {}).join(', '),
+      };
+    }
+  });
+}
+
+export class ClassValidatorValidator implements Validator<ClassType> {
+  canHandle(schema: Schema): schema is ClassType {
+    return typeof (schema as ClassType) === 'function';
+  }
+
+  async validate<
+    T_Schema extends ClassType = ClassType,
+    T_Unvalidated = FromSchemaUnvalidated<T_Schema>,
+    T_Validated = FromSchema<T_Schema>,
+  >(data: T_Unvalidated, schema: T_Schema): Promise<ValidateResult<T_Validated>> {
+    try {
+      await import('reflect-metadata');
+      const { plainToInstance, instanceToPlain } = await import('class-transformer');
+      const { validate } = await import('class-validator');
+
+      // Convert plain data to an instance of the schema class
+      const instance = plainToInstance(schema, data);
+
+      // Validate the instance
+      const errors = await validate(instance as object, { whitelist: true });
+
+      // if undefined, then something went wrong
+      if (!instance && !!data) throw new Error('Failed to convert data to an instance of the schema class');
+
+      if (errors.length === 0) {
+        return { success: true, data: instanceToPlain(instance) as T_Validated };
+      } else {
+        return {
+          success: false,
+          errors: formatErrors(errors),
+        };
+      }
+    } catch (error) {
+      if ((error as Error)?.message?.includes('Cannot find module')) {
+        throw new Error(
+          'Tried to use a class-validator schema in @novu/framework without `class-validator`, `class-transformer` or `reflect-metadata` installed. ' +
+            'Please install it by running `npm install class-validator class-transformer reflect-metadata`.'
+        );
+      }
+      throw error;
+    }
+  }
+
+  async transformToJsonSchema(schema: ClassType): Promise<JsonSchema> {
+    try {
+      // eslint-disable-next-line global-require
+      const { defaultMetadataStorage } = require('class-transformer/cjs/storage');
```

Comment:
@rifont 
I am not able to replace this line with the apropiate import.
This will result in a failing test for get json-schema for nested properties. Would you mind to take a look?

Reply:
The package internals should use ESM for full compatibility, we still transpile the ESM internals _down_ to both a CJS + ESM bundle. We have no plans to drop CJS support.

There's some discussion [here](https://github.com/typestack/class-transformer/issues/563#issuecomment-803262394) about importing from the `class-transformer` internals via the `class-transformer/esm5/storage`. Could we try that?

Reply:
I've added a workaround, to determine which of those should be imported. But idk if this is the way to go.
Thanks for the hint to the discussion btw.
We may should check if they can resolve this shortly, because I think a native import would be much better.

The tests will only succeed currently, when cjs is imported.

Reply:
Ok thanks for confirming. Let's go with a CJS import for now and we can address full ESM compatibility here later.

Could you please add a link to the native import discussion with a `TODO` directive to use the native import when it's available?

Reply:
please take a look at the current implementation, do you see any pitfalls here? should we go with just cjs to begin with?

Reply:
In the end, we'll go with CJS! I have tested on both CJS + ESM successfully.